pkg/gadgets/internal/networktracer/tracer.go

```diff
@@ -33,13 +34,16 @@ import (
 	"github.com/inspektor-gadget/inspektor-gadget/pkg/types"
 )
 
+//go:generate bash -c "source ./clangosflags.sh; go run github.com/cilium/ebpf/cmd/bpf2go -target bpfel -cc clang dispatcher ./bpf/dispatcher.bpf.c -- $CLANG_OS_FLAGS -I./bpf/ -I../socketenricher/bpf"
+
 const (
 	SocketsMapName = "sockets"
 )
 
 type attachment struct {
-	collection *ebpf.Collection
-	perfRd     *perf.Reader
+	dispatcherSpec       *ebpf.CollectionSpec
```

Comment:
Do we really need to keep the spec on the attachment struct? It seems it's only used in `newAttachment`, hence a local variable can work better.